src/Avalonia.Themes.Fluent/Controls/SplitButton.xaml

```diff
@@ -0,0 +1,373 @@
+﻿<Styles xmlns="https://github.com/avaloniaui"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        x:CompileBindings="True">
+  <Design.PreviewWith>
+  </Design.PreviewWith>
+
+  <Styles.Resources>
+    <x:Double x:Key="SplitButtonPrimaryButtonSize">32</x:Double>
+    <x:Double x:Key="SplitButtonSecondaryButtonSize">32</x:Double>
+  </Styles.Resources>
+
+  <Style Selector="SplitButton">
+    <Setter Property="Background" Value="{DynamicResource SplitButtonBackground}" />
+    <Setter Property="Foreground" Value="{DynamicResource SplitButtonForeground}" />
+    <Setter Property="BorderBrush" Value="{DynamicResource SplitButtonBorderBrush}" />
+    <Setter Property="BorderThickness" Value="{DynamicResource SplitButtonBorderThemeThickness}" />
+    <Setter Property="HorizontalAlignment" Value="Left" />
+    <Setter Property="VerticalAlignment" Value="Center" />
+    <!--<Setter Property="FontFamily" Value="{ThemeResource ContentControlThemeFontFamily}" />
+    <Setter Property="FontWeight" Value="Normal" />
+    <Setter Property="FontSize" Value="{ThemeResource ControlContentThemeFontSize}" />
+    <Setter Property="UseSystemFocusVisuals" Value="True" />
+    <Setter Property="FocusVisualMargin" Value="-3" />-->
+    <Setter Property="KeyboardNavigation.IsTabStop" Value="True" />
+    <Setter Property="Padding" Value="{DynamicResource ButtonPadding}" />
+    <Setter Property="CornerRadius" Value="{DynamicResource ControlCornerRadius}" />
+    <Setter Property="Template">
+      <Setter.Value>
+        <ControlTemplate>
+          <!-- A Border was added here since Avalonia does not support Background/Border/CornerRadius on Grid directly -->
+          <!-- ClipToBounds added as a temporary work-around for: https://github.com/AvaloniaUI/Avalonia/issues/2105 -->
+          <Border Background="Transparent"
+                  CornerRadius="{TemplateBinding CornerRadius}"
+                  ClipToBounds="True">
+            <Border.Styles>
+              <!-- Override the style of the inner buttons so that they don't affect background/foreground/border colors -->
+              <Style Selector="Button">
```

Comment:
I would recommend extracting this style out of template and using selector by name.
It will greatly simplify template replacing, if developer needs some minor structure changes without need to copy paste buttons styles as well.

Reply:
Or better to move this style outside with this selector:
```
<Style Selector="SplitButton /template/ Button>
```

Reply:
Same with Button:disabled below

Reply:
> Same with Button:disabled below

Can be moved out, I already considered that earlier. The rest of it might require some further testing. That said, again, this is following WinUI. Do you make such large changes for the Fluent theme from upstream? It just makes it much harder to keep styles synced which is actually the most relevant. Code can be allowed to diverge more without too much trouble.

Reply:
We are already making so many changes in the Style, I think maintaining easy compatibility with upstream Fluent / Fluentv2 changes doesn't matter now. So that decision seems to have been made -- it just requires much more work for future synchronizations.

Reply:
Should be resolved: the disabled setters were pulled out and placed within the disabled state section. That makes it a bit easier to read actually.

https://github.com/AvaloniaUI/Avalonia/blob/c2cebb32fa5423ba174b9a87817f11a3c3fa40a3/src/Avalonia.Themes.Fluent/Controls/SplitButton.xaml#L154-L158

Reply:
Completely reworked in latest commits. Button properties are now set directly in the various states and background borders (formerly rectangles) removed.